gap/elements/boolmat.gi

```diff
@@ -265,8 +266,8 @@ function(x)
   local n, contained, row, i, j, k;
 
   n := Length(x![1]);
-  for i in [1 .. n - 1] do
-    for j in [i + 1 .. n] do
+  for i in [1 .. n] do
+    for j in Concatenation([1 .. i - 1], [i + 1 .. n]) do
```

Comment:
I don't know if this'll make much difference, but might it not be better to just check if j = i inside the loop here?